tests/conftest.py

```diff
@@ -4720,3 +4720,38 @@ def patch_consumer_toolbox_with_secret():
 
     log.info("Switching back to the initial cluster context")
     config.switch_ctx(restore_ctx_index)
+
+
+@pytest.fixture(scope="function", autouse=True)
+def switch_to_provider_for_test(request):
+    """
+    Switch to provider cluster as required by the test. Applicable for Managed Services only if
+    the marker 'runs_on_provider' is added in the test.
+
+    """
+    switched_to_provider = False
+    current_cluster = config.cluster_ctx
+    if (
+        request.node.get_closest_marker("runs_on_provider")
+        and config.multicluster
+        and current_cluster.ENV_DATA.get("platform", "").lower()
+        in constants.MANAGED_SERVICE_PLATFORMS
+    ):
+        for cluster in config.clusters:
+            if cluster.ENV_DATA.get("cluster_type") == "provider":
+                provider_cluster = cluster
+                log.debug("Switching to the provider cluster context")
+                config.switch_ctx(provider_cluster.MULTICLUSTER["multicluster_index"])
+                switched_to_provider = True
+                break
```

Comment:
There is a new function `switch_to_provider` introduced in [PR#5541](https://github.com/red-hat-storage/ocs-ci/pull/5541/files#diff-1aebcb25bdbc9860fbc89fc474c3ddc4c6ac92c93071487c9cf610a35b773655R268) which might be used here, but the PR is not yet merged - so it is mainly suggestion for future improvement.

Reply:
Thanks @dahorak . Enhancements will be done after merging #5541

Reply:
@fbalak Added # TODO.